crengine/src/private/lvfreetypeface.cpp

```diff
@@ -1506,20 +1524,21 @@ lUInt16 LVFreeTypeFace::measureText(const lChar32 *text,
                 flags[t] = GET_CHAR_FLAGS(text[t]);
                 // It seems each soft-hyphen is in its own cluster, of length 1 and width 0,
                 // so HarfBuzz must already deal correctly with soft-hyphens.
-                if (cur_width == prev_width) {
+                if (cur_width_26_6 == prev_width_26_6) {
                     // But if there is no advance (this happens with soft-hyphens),
                     // flag it and don't add any letter spacing.
                     flags[t] |= LCHAR_IS_CLUSTER_TAIL;
                 }
                 else {
-                    cur_width += letter_spacing; // only between clusters/graphemes
-                }            
+                    cur_width_26_6 += letter_spacing_26_6; // only between clusters/graphemes
+                }
             }
+            int cur_width = FONT_METRIC_TO_PX(cur_width_26_6);
             widths[t] = cur_width;
```

Comment:
I'm a bit saddened by all this _26_6 added noise in my painfully crafted harfbuzz code :)
But ok, your latest commit makes it a bit more contained.

Hard to get back into that code.
I am right in thinking this is the only place where there is the translation between internal 26_6 stuff and the final returned widths[] in pixels ?
No risk of accumulated rounding errors with working internally with 26_6 and adding rounded TO_PX to the widths ? (I guess it's safe as we return cumulated widths, just asking in case you wondered the same thing and answered similarly :)

Reply:
Yes, in short, this was introduced to get rid of round-off errors.